File: 3812-smallest-palindromic-rearrangement-i/smallest-palindromic-rearrangement-i.cpp
class Solution {
public:
    string smallestPalindrome(string s) {
        int n = s.length();
        vector<int> arr (26,0);

        for(char c : s) {
            arr[c - 'a']++;
        }

        string res = "";
        int idx = -1;
        for (int i = 0; i<26; i++) {
            if(arr[i] % 2 == 1) {
                idx = i;
            } 
            res += string (arr[i]/2 , char(i + 'a'));
        }

        string temp = res;

        if(idx != -1) {
            res.push_back(idx + 'a');
        }

        reverse(temp.begin(),temp.end());

        res += temp;

        return res;
    }
};

File: 3812-smallest-palindromic-rearrangement-i/smallest-palindromic-rearrangement-i.py
class Solution:
    def smallestPalindrome(self, s: str) -> str:
        n = len(s)
        arr = [0] * 26

        for c in s :
            arr[ord(c) - ord('a')] += 1

        idx = -1
        st = ""

        for i in range (26) :
            if arr[i] % 2 == 1 : idx = i
            st += chr(i + 97) * (arr[i] // 2)
        
        temp = st

        if idx != -1 : st += chr(idx + 97)

        st += temp[::-1]
        
        return st        